pages/api/enroll/child.ts

```diff
@@ -17,6 +19,22 @@ export default async function handle(
     req: NextApiRequest,
     res: NextApiResponse,
 ): Promise<void> {
+    const session = await getSession({ req });
+
+    // If there is no session or the user is not a parent
+    if (!session || session.role !== roles.PARENT) {
+        return ResponseUtil.returnUnauthorized(res, "Unauthorized");
```

Comment:
nit: could use a more descriptive error message - maybe something like `only users with the PARENT role can access this resource`

Reply:
good idea!

Reply:
Fixed!